packages/fx-core/src/component/driver/deploy/azure/baseDeployDriver.ts

```diff
@@ -23,26 +27,25 @@ export abstract class BaseDeployDriver extends BaseDeployStepDriver {
     dryRun: asOptional(asBoolean),
   });
 
-  async run(): Promise<Map<string, string>> {
+  async run(): Promise<ExecutionResult> {
     await this.context.logProvider.debug("start deploy process");
 
-    const deployArgs = BaseDeployDriver.asDeployArgs(this.args, this.helpLink);
-    // if working directory not set, use current working directory
-    deployArgs.workingDirectory = deployArgs.workingDirectory ?? "./";
-    // if working dir is not absolute path, then join the path with project path
-    this.workingDirectory = path.isAbsolute(deployArgs.workingDirectory)
-      ? deployArgs.workingDirectory
-      : path.join(this.workingDirectory, deployArgs.workingDirectory);
-    // if distribution path is not absolute path, then join the path with project path
-    this.distDirectory = path.isAbsolute(deployArgs.distributionPath)
-      ? deployArgs.distributionPath
-      : path.join(this.workingDirectory, deployArgs.distributionPath);
-    this.dryRun = deployArgs.dryRun ?? false;
-    // call real deploy
-    await this.wrapErrorHandler(async () => {
-      await this.deploy(deployArgs);
+    return await this.wrapErrorHandler(async () => {
+      const deployArgs = BaseDeployDriver.asDeployArgs(this.args, this.helpLink);
+      // if working directory not set, use current working directory
+      deployArgs.workingDirectory = deployArgs.workingDirectory ?? "./";
+      // if working dir is not absolute path, then join the path with project path
+      this.workingDirectory = path.isAbsolute(deployArgs.workingDirectory)
+        ? deployArgs.workingDirectory
+        : path.join(this.workingDirectory, deployArgs.workingDirectory);
+      // if distribution path is not absolute path, then join the path with project path
+      this.distDirectory = path.isAbsolute(deployArgs.distributionPath)
+        ? deployArgs.distributionPath
+        : path.join(this.workingDirectory, deployArgs.distributionPath);
```

Comment:
It looks a little weird that this.distributionPath depends on the value of this.workingDirectory?

Reply:
`distributionPath` is always the same or a subdirectory of the working director. Because in most cases the output of build is in the working directory.